aiosmtpd/smtp.py

```diff
@@ -438,7 +438,13 @@ def smtp_DATA(self, arg):
                 'mail_options': self.mail_options,
                 'rcpt_options': self.rcpt_options,
             }
-        status = self.event_handler.process_message(*args, **kwargs)
+        kwargs.update({'loop': self.loop})
+        if asyncio.iscoroutinefunction(self.event_handler.process_message):
+            kwargs['loop'] = self.loop
```

Comment:
I think this line is redundant now right?  But actually, I prefer this style over the one in line 441.  Can you just move this up out of the if-else?